### PR TITLE
error handling in bucket creation ui

### PR DESCRIPTION
**Pull request by prakashsvmx**
error handling in bucket creation ui

Fixes 2459 

Test cases
- Policy permissions
- MinIO not reachable

image

e.g: Create a bucket with a pattern which does not allow the user to to create ( using iam policy)

### CODE REVIEW

File: web-app/src/screens/Console/Buckets/ListBuckets/AddBucket/addBucketsSlice.ts [hunk 1]
```diff
@@ -36,6 +36,7 @@ interface AddBucketState {
   navigateTo: string;
   excludeFolders: boolean;
   excludedPrefixes: string;
+  error: any;
```
Comment (bayasdev) [inline web-app/src/screens/Console/Buckets/ListBuckets/AddBucket/addBucketsSlice.ts:39]: @prakashsvmx does this have to be of type any?

**Review by bayasdev — CHANGES_REQUESTED** [review]

File: web-app/src/screens/Console/Buckets/ListBuckets/AddBucket/addBucketsSlice.ts (hunk 1 above)
Comment (prakashsvmx) [reply web-app/src/screens/Console/Buckets/ListBuckets/AddBucket/addBucketsSlice.ts:39]: I think it is the same type in the other component and expects it that way.  i tried to use `ApiError` . 
let me double check .

**Review by reivaj05 — DISMISSED** [review]

File: web-app/src/screens/Console/Buckets/ListBuckets/AddBucket/addBucketsSlice.ts (hunk 1 above)
Comment (prakashsvmx) [reply web-app/src/screens/Console/Buckets/ListBuckets/AddBucket/addBucketsSlice.ts]: i have updated the type. thank you for pointing this out.

**Review by bayasdev — APPROVED** [review]

File: web-app/src/screens/Console/Buckets/ListBuckets/AddBucket/addBucketsSlice.ts (hunk 1 above)
Comment (bayasdev) [reply web-app/src/screens/Console/Buckets/ListBuckets/AddBucket/addBucketsSlice.ts]: Thanks @prakashsvmx

**Review by cesnietor — APPROVED** [review]

**Comment by cesnietor** [comment]
@prakashsvmx  could you please take a look at the failing tests, seem unrelated but all permission ones are failing

**Comment by prakashsvmx** [comment]
@cesnietor  something broken in testcafe/chrome versions in CI. there is not release/update from testcafe. we need to wait for stable release. 

i tried with `rc` release  Ref: https://github.com/minio/console/pull/3457

I will investigate when there is a new release of testcafe.

**Review by bexsoft — APPROVED** [review]

**Review by reivaj05 — APPROVED** [review]

**Pull request merged**